x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TimeseriesFieldTypeHelper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rollup.v2;
+
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.index.mapper.TimeSeriesParams;
+import org.elasticsearch.indices.IndicesService;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.elasticsearch.index.mapper.TimeSeriesParams.TIME_SERIES_DIMENSION_PARAM;
+import static org.elasticsearch.index.mapper.TimeSeriesParams.TIME_SERIES_METRIC_PARAM;
+
+class TimeseriesFieldTypeHelper {
+
+    private final MapperService mapperService;
+    private final String timestampField;
+
+    private TimeseriesFieldTypeHelper(final MapperService mapperService, final String timestampField) {
+        this.mapperService = mapperService;
+        this.timestampField = timestampField;
+    }
+
+    public boolean isTimeSeriesLabel(final String field, final Map<String, ?> unused) {
+        final MappedFieldType fieldType = mapperService.mappingLookup().getFieldType(field);
+        return fieldType != null
+            && (timestampField.equals(field) == false)
+            && (fieldType.isAggregatable())
```

Comment:
This is basically to avoid fields of type `text` which is not aggregatable...while it would work with fields of type `keyword` which are aggregatable.

Reply:
If I remove that condition the tests results in an errors saying something like "you need to use aggregatable fields or use <some API I don't remember> to support non-aggregatable fields

Reply:
cool thanks for the explanation. I had missed that

Reply:
This actually means that we cannot have a `text` field as label?